diff -Naur a/init/main.c b/init/main.c
--- a/init/main.c	2026-02-08 22:03:27.000000000 +0100
+++ b/init/main.c	2026-02-20 11:15:49.953844302 +0100
@@ -1631,7 +1631,8 @@
 			return 0;
 	}
 
-	if (!try_to_run_init_process("/sbin/init") ||
+	if (!try_to_run_init_process("/usr/bin/init") ||
+	    !try_to_run_init_process("/sbin/init") ||
 	    !try_to_run_init_process("/etc/init") ||
 	    !try_to_run_init_process("/bin/init") ||
 	    !try_to_run_init_process("/bin/sh"))
