diff -Naur a/meson.build b/meson.build
--- a/meson.build	2023-07-28 10:26:32.000000000 +0200
+++ b/meson.build	2024-12-14 13:12:55.844937175 +0100
@@ -4007,7 +4007,7 @@
         link_with : [libshared],
         dependencies : userspace,
         install_rpath : rootpkglibdir,
-        install : (conf.get('HAVE_SYSV_COMPAT') == 1),
+        install : true,
         install_dir : rootlibexecdir)
 
 public_programs += executable(
diff -Naur a/src/basic/path-util.h b/src/basic/path-util.h
--- a/src/basic/path-util.h	2023-07-28 10:26:32.000000000 +0200
+++ b/src/basic/path-util.h	2024-12-14 13:16:42.723468942 +0100
@@ -11,8 +11,8 @@
 #include "strv.h"
 #include "time-util.h"
 
-#define PATH_SPLIT_SBIN_BIN(x) x "sbin:" x "bin"
-#define PATH_SPLIT_SBIN_BIN_NULSTR(x) x "sbin\0" x "bin\0"
+#define PATH_SPLIT_SBIN_BIN(x) x "bin:" x "sbin"
+#define PATH_SPLIT_SBIN_BIN_NULSTR(x) x "bin\0" x "sbin\0"
 
 #define PATH_NORMAL_SBIN_BIN(x) x "bin"
 #define PATH_NORMAL_SBIN_BIN_NULSTR(x) x "bin\0"
diff -Naur a/src/basic/unit-def.h b/src/basic/unit-def.h
--- a/src/basic/unit-def.h	2023-07-28 10:26:32.000000000 +0200
+++ b/src/basic/unit-def.h	2024-12-14 13:17:39.560602227 +0100
@@ -183,8 +183,8 @@
 
 typedef enum SwapState {
         SWAP_DEAD,
-        SWAP_ACTIVATING,               /* /sbin/swapon is running, but the swap not yet enabled. */
-        SWAP_ACTIVATING_DONE,          /* /sbin/swapon is running, and the swap is done. */
+        SWAP_ACTIVATING,               /* /usr/sbin/swapon is running, but the swap not yet enabled. */
+        SWAP_ACTIVATING_DONE,          /* /usr/sbin/swapon is running, and the swap is done. */
         SWAP_ACTIVE,
         SWAP_DEACTIVATING,
         SWAP_DEACTIVATING_SIGTERM,
diff -Naur a/src/core/swap.c b/src/core/swap.c
--- a/src/core/swap.c	2023-07-28 10:26:32.000000000 +0200
+++ b/src/core/swap.c	2024-12-14 13:19:53.088364265 +0100
@@ -821,7 +821,7 @@
                 }
         }
 
-        r = exec_command_set(s->control_command, "/sbin/swapon", "--fixpgsz", NULL);
+        r = exec_command_set(s->control_command, "/usr/sbin/swapon", "--fixpgsz", NULL);
         if (r < 0)
                 goto fail;
 
@@ -859,7 +859,7 @@
         s->control_command = s->exec_command + SWAP_EXEC_DEACTIVATE;
 
         r = exec_command_set(s->control_command,
-                             "/sbin/swapoff",
+                             "/usr/sbin/swapoff",
                              s->what,
                              NULL);
         if (r < 0)
@@ -1379,7 +1379,7 @@
 
                 /* Dispatch this before we dispatch SIGCHLD, so that
                  * we always get the events from /proc/swaps before
-                 * the SIGCHLD of /sbin/swapon. */
+                 * the SIGCHLD of /usr/sbin/swapon. */
                 r = sd_event_source_set_priority(m->swap_event_source, SD_EVENT_PRIORITY_NORMAL-10);
                 if (r < 0) {
                         log_error_errno(r, "Failed to change /proc/swaps priority: %m");
diff -Naur a/src/resolve/meson.build b/src/resolve/meson.build
--- a/src/resolve/meson.build	2023-07-28 10:26:32.000000000 +0200
+++ b/src/resolve/meson.build	2024-12-14 13:21:17.485722704 +0100
@@ -138,56 +138,6 @@
 
 ############################################################
 
-test_resolve_base = {
-        'link_with' : [libsystemd_resolve_core, libshared],
-        'dependencies' : [lib_openssl_or_gcrypt, libm],
-}
-
-tests += [
-        {
-                'sources' : files('test-resolve-tables.c'),
-                'base' : test_resolve_base,
-        },
-        {
-                'sources' : files('test-dns-packet.c'),
-                'base' : test_resolve_base,
-        },
-        {
-                'sources' : files(
-                        'test-resolved-etc-hosts.c',
-                        'resolved-etc-hosts.c',
-                ),
-                'base' : test_resolve_base,
-        },
-        {
-                'sources' : files('test-resolved-packet.c'),
-                'base' : test_resolve_base,
-        },
-        {
-                'sources' : files('test-dnssec.c'),
-                'base' : test_resolve_base,
-                'condition' : 'HAVE_OPENSSL_OR_GCRYPT',
-        },
-        {
-                'sources' : files('test-dnssec-complex.c'),
-                'base' : test_resolve_base,
-                'type' : 'manual',
-        },
-        {
-                'sources' : [
-                        files('test-resolved-stream.c'),
-                        basic_dns_sources,
-                        systemd_resolved_sources,
-                ],
-                'dependencies' : [
-                        lib_openssl_or_gcrypt,
-                        libm,
-                        systemd_resolved_dependencies,
-                ],
-                'includes' : resolve_includes,
-        },
-]
-
 fuzzer_resolve_base = {
         'link_with' : [libsystemd_resolve_core, libshared],
         'dependencies' : [lib_openssl_or_gcrypt, libm],
diff -Naur a/units/modprobe@.service b/units/modprobe@.service
--- a/units/modprobe@.service	2023-07-28 10:26:32.000000000 +0200
+++ b/units/modprobe@.service	2024-12-14 13:22:06.446123836 +0100
@@ -17,4 +17,4 @@
 
 [Service]
 Type=oneshot
-ExecStart=-/sbin/modprobe -abq %I
+ExecStart=-modprobe -abq %I
