diff -Naur a/src/basic/path-util.h b/src/basic/path-util.h
--- a/src/basic/path-util.h	2025-10-13 17:01:55.000000000 +0200
+++ b/src/basic/path-util.h	2025-11-16 19:07:44.851016080 +0100
@@ -3,8 +3,8 @@
 
 #include "forward.h"
 
-#define PATH_SPLIT_BIN(x) x "sbin:" x "bin"
-#define PATH_SPLIT_BIN_NULSTR(x) x "sbin\0" x "bin\0"
+#define PATH_SPLIT_BIN(x) x "bin:" x "sbin"
+#define PATH_SPLIT_BIN_NULSTR(x) x "bin\0" x "sbin\0"
 
 #define PATH_MERGED_BIN(x) x "bin"
 #define PATH_MERGED_BIN_NULSTR(x) x "bin\0"
diff -Naur a/src/basic/unit-def.h b/src/basic/unit-def.h
--- a/src/basic/unit-def.h	2025-10-13 17:01:55.000000000 +0200
+++ b/src/basic/unit-def.h	2025-11-16 19:08:39.156662599 +0100
@@ -185,8 +185,8 @@
 
 typedef enum SwapState {
         SWAP_DEAD,
-        SWAP_ACTIVATING,               /* /sbin/swapon is running, but the swap not yet enabled. */
-        SWAP_ACTIVATING_DONE,          /* /sbin/swapon is running, and the swap is done. */
+        SWAP_ACTIVATING,               /* /usr/sbin/swapon is running, but the swap not yet enabled. */
+        SWAP_ACTIVATING_DONE,          /* /usr/sbin/swapon is running, and the swap is done. */
         SWAP_ACTIVE,
         SWAP_DEACTIVATING,
         SWAP_DEACTIVATING_SIGTERM,
diff -Naur a/src/core/swap.c b/src/core/swap.c
--- a/src/core/swap.c	2025-10-13 17:01:55.000000000 +0200
+++ b/src/core/swap.c	2025-11-16 19:09:56.548576552 +0100
@@ -781,7 +781,7 @@
                 }
         }
 
-        r = exec_command_set(s->control_command, "/sbin/swapon", "--fixpgsz", NULL);
+        r = exec_command_set(s->control_command, "/usr/sbin/swapon", "--fixpgsz", NULL);
         if (r < 0) {
                 log_unit_warning_errno(UNIT(s), r, "Failed to initialize swapon command line: %m");
                 goto fail;
@@ -826,7 +826,7 @@
         s->control_command = s->exec_command + SWAP_EXEC_DEACTIVATE;
 
         r = exec_command_set(s->control_command,
-                             "/sbin/swapoff",
+                             "/usr/sbin/swapoff",
                              s->what,
                              NULL);
         if (r < 0) {
@@ -1330,7 +1330,7 @@
 
                 /* Dispatch this before we dispatch SIGCHLD, so that
                  * we always get the events from /proc/swaps before
-                 * the SIGCHLD of /sbin/swapon. */
+                 * the SIGCHLD of /usr/sbin/swapon. */
                 r = sd_event_source_set_priority(m->swap_event_source, EVENT_PRIORITY_SWAP_TABLE);
                 if (r < 0) {
                         log_error_errno(r, "Failed to change /proc/swaps priority: %m");
diff -Naur a/src/kernel-install/50-depmod.install b/src/kernel-install/50-depmod.install
--- a/src/kernel-install/50-depmod.install	2025-10-13 17:01:55.000000000 +0200
+++ b/src/kernel-install/50-depmod.install	2025-11-16 19:14:23.699002350 +0100
@@ -23,11 +23,11 @@
 COMMAND="${1:?}"
 KERNEL_VERSION="${2:?}"
 
-[ -w "/lib/modules" ] || exit 0
+[ -w "/usr/lib/modules" ] || exit 0
 
 case "$COMMAND" in
     add)
-        [ -d "/lib/modules/$KERNEL_VERSION/kernel" ] || exit 0
+        [ -d "/usr/lib/modules/$KERNEL_VERSION/kernel" ] || exit 0
         command -v depmod >/dev/null || exit 0
         [ "$KERNEL_INSTALL_VERBOSE" -gt 0 ] && echo "+depmod -a $KERNEL_VERSION"
         exec depmod -a "$KERNEL_VERSION"
@@ -40,19 +40,19 @@
         fi
 
         [ "$KERNEL_INSTALL_VERBOSE" -gt 0 ] && \
-            echo "Removing /lib/modules/${KERNEL_VERSION}/modules.dep and associated files"
+            echo "Removing /usr/lib/modules/${KERNEL_VERSION}/modules.dep and associated files"
         exec rm -f \
-            "/lib/modules/$KERNEL_VERSION/modules.alias" \
-            "/lib/modules/$KERNEL_VERSION/modules.alias.bin" \
-            "/lib/modules/$KERNEL_VERSION/modules.builtin.bin" \
-            "/lib/modules/$KERNEL_VERSION/modules.builtin.alias.bin" \
-            "/lib/modules/$KERNEL_VERSION/modules.dep" \
-            "/lib/modules/$KERNEL_VERSION/modules.dep.bin" \
-            "/lib/modules/$KERNEL_VERSION/modules.devname" \
-            "/lib/modules/$KERNEL_VERSION/modules.softdep" \
-            "/lib/modules/$KERNEL_VERSION/modules.weakdep" \
-            "/lib/modules/$KERNEL_VERSION/modules.symbols" \
-            "/lib/modules/$KERNEL_VERSION/modules.symbols.bin"
+            "/usr/lib/modules/$KERNEL_VERSION/modules.alias" \
+            "/usr/lib/modules/$KERNEL_VERSION/modules.alias.bin" \
+            "/usr/lib/modules/$KERNEL_VERSION/modules.builtin.bin" \
+            "/usr/lib/modules/$KERNEL_VERSION/modules.builtin.alias.bin" \
+            "/usr/lib/modules/$KERNEL_VERSION/modules.dep" \
+            "/usr/lib/modules/$KERNEL_VERSION/modules.dep.bin" \
+            "/usr/lib/modules/$KERNEL_VERSION/modules.devname" \
+            "/usr/lib/modules/$KERNEL_VERSION/modules.softdep" \
+            "/usr/lib/modules/$KERNEL_VERSION/modules.weakdep" \
+            "/usr/lib/modules/$KERNEL_VERSION/modules.symbols" \
+            "/usr/lib/modules/$KERNEL_VERSION/modules.symbols.bin"
         ;;
     *)
         exit 0
diff -Naur a/test/units/TEST-35-LOGIN.sh b/test/units/TEST-35-LOGIN.sh
--- a/test/units/TEST-35-LOGIN.sh	2025-10-13 17:01:55.000000000 +0200
+++ b/test/units/TEST-35-LOGIN.sh	2025-11-16 19:15:26.806420321 +0100
@@ -351,7 +351,7 @@
 [Service]
 Type=simple
 ExecStart=
-ExecStart=-/sbin/agetty --autologin logind-test-user --noclear %I $TERM
+ExecStart=-/usr/sbin/agetty --autologin logind-test-user --noclear %I $TERM
 Restart=no
 EOF
     systemctl daemon-reload
diff -Naur a/tmpfiles.d/home.conf b/tmpfiles.d/home.conf
--- a/tmpfiles.d/home.conf	2025-10-13 17:01:55.000000000 +0200
+++ b/tmpfiles.d/home.conf	2025-11-16 19:15:43.416612261 +0100
@@ -7,5 +7,4 @@
 
 # See tmpfiles.d(5) for details.
 
-Q /home 0755 - - -
 q /srv 0755 - - -
diff -Naur a/units/console-getty.service.in b/units/console-getty.service.in
--- a/units/console-getty.service.in	2025-10-13 17:01:55.000000000 +0200
+++ b/units/console-getty.service.in	2025-11-16 19:16:28.633134381 +0100
@@ -20,7 +20,7 @@
 ConditionPathExists=/dev/console
 
 [Service]
-ExecStart=-/sbin/agetty --noreset --noclear --issue-file=/etc/issue:/etc/issue.d:/run/issue.d:/usr/lib/issue.d --keep-baud 115200,57600,38400,9600 - ${TERM}
+ExecStart=-/usr/sbin/agetty --noreset --noclear --issue-file=/etc/issue:/etc/issue.d:/run/issue.d:/usr/lib/issue.d --keep-baud 115200,57600,38400,9600 - ${TERM}
 Type=idle
 Restart=always
 UtmpIdentifier=cons
diff -Naur a/units/container-getty@.service.in b/units/container-getty@.service.in
--- a/units/container-getty@.service.in	2025-10-13 17:01:55.000000000 +0200
+++ b/units/container-getty@.service.in	2025-11-16 19:16:49.488375030 +0100
@@ -25,7 +25,7 @@
 Before=rescue.service
 
 [Service]
-ExecStart=-/sbin/agetty --noreset --noclear --issue-file=/etc/issue:/etc/issue.d:/run/issue.d:/usr/lib/issue.d - ${TERM}
+ExecStart=-/usr/sbin/agetty --noreset --noclear --issue-file=/etc/issue:/etc/issue.d:/run/issue.d:/usr/lib/issue.d - ${TERM}
 Type=idle
 Restart=always
 RestartSec=0
diff -Naur a/units/getty@.service.in b/units/getty@.service.in
--- a/units/getty@.service.in	2025-10-13 17:01:55.000000000 +0200
+++ b/units/getty@.service.in	2025-11-16 19:17:54.674126629 +0100
@@ -34,7 +34,7 @@
 ConditionPathExists=/dev/tty0
 
 [Service]
-ExecStart=-/sbin/agetty --noreset --noclear --issue-file=/etc/issue:/etc/issue.d:/run/issue.d:/usr/lib/issue.d - ${TERM}
+ExecStart=-/usr/sbin/agetty --noreset --noclear --issue-file=/etc/issue:/etc/issue.d:/run/issue.d:/usr/lib/issue.d - ${TERM}
 Type=idle
 Restart=always
 RestartSec=0
diff -Naur a/units/kmod-static-nodes.service.in b/units/kmod-static-nodes.service.in
--- a/units/kmod-static-nodes.service.in	2025-10-13 17:01:55.000000000 +0200
+++ b/units/kmod-static-nodes.service.in	2025-11-16 19:18:18.990406804 +0100
@@ -12,7 +12,7 @@
 DefaultDependencies=no
 Before=sysinit.target systemd-tmpfiles-setup-dev-early.service
 ConditionCapability=CAP_SYS_MODULE
-ConditionFileNotEmpty=/lib/modules/%v/modules.devname
+ConditionFileNotEmpty=/usr/lib/modules/%v/modules.devname
 ConditionPathExists={{KMOD}}
 
 [Service]
diff -Naur a/units/modprobe@.service b/units/modprobe@.service
--- a/units/modprobe@.service	2025-10-13 17:01:55.000000000 +0200
+++ b/units/modprobe@.service	2025-11-16 19:18:48.994746217 +0100
@@ -32,4 +32,4 @@
 
 [Service]
 Type=oneshot
-ExecStart=-/sbin/modprobe -abq %i
+ExecStart=-/usr/sbin/modprobe -abq %i
diff -Naur a/units/serial-getty@.service.in b/units/serial-getty@.service.in
--- a/units/serial-getty@.service.in	2025-10-13 17:01:55.000000000 +0200
+++ b/units/serial-getty@.service.in	2025-11-16 19:19:30.287197392 +0100
@@ -30,7 +30,7 @@
 Before=rescue.service
 
 [Service]
-ExecStart=-/sbin/agetty --noreset --noclear --issue-file=/etc/issue:/etc/issue.d:/run/issue.d:/usr/lib/issue.d --keep-baud 115200,57600,38400,9600 - ${TERM}
+ExecStart=-/usr/sbin/agetty --noreset --noclear --issue-file=/etc/issue:/etc/issue.d:/run/issue.d:/usr/lib/issue.d --keep-baud 115200,57600,38400,9600 - ${TERM}
 Type=idle
 Restart=always
 UtmpIdentifier=%I
