diff -Naur a/src/Makefile.am b/src/Makefile.am
--- a/src/Makefile.am	2014-10-30 04:29:54.000000000 +0100
+++ b/src/Makefile.am	2024-12-11 15:08:09.800126287 +0100
@@ -95,7 +95,7 @@
 
 libsox_la_CFLAGS = @WARN_CFLAGS@
 libsox_la_LDFLAGS = @APP_LDFLAGS@ -version-info @SHLIB_VERSION@ \
-  -export-symbols-regex '^(sox_.*|lsx_(check_read_params|(close|open)_dllibrary|(debug(_more|_most)?|fail|report|warn)_impl|eof|fail_errno|filelength|find_(enum_(text|value)|file_extension)|getopt(_init)?|lpc10_(create_(de|en)coder_state|(de|en)code)|raw(read|write)|read(_b_buf|buf|chars)|realloc|rewind|seeki|sigfigs3p?|strcasecmp|tell|unreadb|write(b|_b_buf|buf|s)))$$'
+  -export-symbols-regex '^(sox_.*|lsx_(error|flush|check_read_params|(close|open)_dllibrary|(debug(_more|_most)?|fail|report|warn)_impl|eof|fail_errno|filelength|find_(enum_(text|value)|file_extension)|getopt(_init)?|lpc10_(create_(de|en)coder_state|(de|en)code)|raw(read|write)|read(_b_buf|buf|chars)|realloc|rewind|seeki|sigfigs3p?|strcasecmp|tell|unreadb|write(b|_b_buf|buf|s)))$$'
 
 if HAVE_WIN32_LTDL
   libsox_la_SOURCES += win32-ltdl.c win32-ltdl.h
diff -Naur a/src/sox_sample_test.h b/src/sox_sample_test.h
--- a/src/sox_sample_test.h	2012-01-23 23:27:33.000000000 +0100
+++ b/src/sox_sample_test.h	2024-12-11 15:08:54.047150312 +0100
@@ -20,6 +20,7 @@
 #endif
 #include <assert.h>
 #include "sox.h"
+#include <math.h>
 
 #define TEST_UINT(bits) \
   uint##bits = 0; \
