diff --git a/scripts/mybld.in b/scripts/mybld.in
index d421bd8..6037a0e 100755
--- a/scripts/mybld.in
+++ b/scripts/mybld.in
@@ -631,7 +631,7 @@ DEVEL="cd /$GIT \
 $PAUSE "
 
 if [ -z $MANUAL ]; then
-  DEVEL+="&& cards create --no-meta"
+  DEVEL+="&& cards create"
   if [ ! -z ${PACKAGES_LIST} ]; then
     if [ "$DIFF" == "yes" ];then
       DEVEL="$DEVEL $COLLECTION --dry"
diff --git a/src/create.cxx b/src/create.cxx
index c32bcc8..85fa7be 100644
--- a/src/create.cxx
+++ b/src/create.cxx
@@ -299,6 +299,7 @@ void create::run()
         removeFile (m_root, "/.META");
         removeFile (m_root, "/.PRE");
         removeFile (m_root, "/.POST");
+        removeFile (m_root, "/.README");
 	    m_listOfPackages[package.first] = package.second;
     }
 	postRun();
