diff --git a/src/create.cxx b/src/create.cxx
index fa67470..6c9804f 100644
--- a/src/create.cxx
+++ b/src/create.cxx
@@ -13,6 +13,11 @@ create::create(CardsArgumentParser& argParser)
     m_portsDir = getcwd(pwd,MAXPATHLEN);
 
     parseArguments();
+    if (!checkFileExist(m_root + "tmp/setup")) {
+        m_actualError = cards::ERROR_ENUM_NOT_IN_CHROOT;
+        treatErrors(m_root + "tmp/setup");
+    }
+
     m_tree = m_pkgfile.getListOfPackages();
 
     if (isACollection())
@@ -94,6 +99,10 @@ void create::treatErrors(const std::string& s) const
     case ERROR_ENUM_CANNOT_CHANGE_DIRECTORY:
         throw RunTimeErrorWithErrno(_("could not change directory ") + s);
         break;
+    case ERROR_ENUM_NOT_IN_CHROOT:
+        throw RunTimeErrorWithErrno(_("You are not in chroot mode. The file ") + s
+        + _(" is not found"));
+        break;
     }
 }
 void create::checkBinaries()
diff --git a/src/enum.h b/src/enum.h
index 4eb27f3..bcc4bf4 100644
--- a/src/enum.h
+++ b/src/enum.h
@@ -45,6 +45,7 @@ enum ErrorEnum {
 	ERROR_ENUM_PKGADD_CONFIG_UNKNOWN_ACTION,
 	ERROR_ENUM_PKGADD_CONFIG_UNKNOWN_EVENT,
 	ERROR_ENUM_CANNOT_COMPILE_REGULAR_EXPRESSION,
+	ERROR_ENUM_NOT_IN_CHROOT,
 	ERROR_ENUM_CANNOT_GENERATE_LEVEL,
 	ERROR_ENUM_NOT_INSTALL_PACKAGE_NEITHER_PACKAGE_FILE,
 	ERROR_ENUM_PRIVATE_KEY_PATH_NOT_DEFINE,
