diff --git a/scripts/mybld.in b/scripts/mybld.in
index 6efef15..2b1930d 100755
--- a/scripts/mybld.in
+++ b/scripts/mybld.in
@@ -388,7 +388,7 @@ find_set () {
   do
      for n in $c/*/Pkgfile
      do
-         set=$(. $n; echo $set)
+         set=($(. $n; echo "${set[@]}"))
          [ -z $set ] && continue
          for s in ${set[@]}
          do
diff --git a/scripts/pkgmk.in b/scripts/pkgmk.in
index 26ceb8c..d692ccc 100755
--- a/scripts/pkgmk.in
+++ b/scripts/pkgmk.in
@@ -815,7 +815,7 @@ add_meta_to_archive() {
 			echo "T$i" >> .META
 		done
 	fi
-	if [ ! -z "$conflicts"]; then
+	if [ ! -z "$conflicts" ]; then
 		info "Adding Conflict packages to Archive $NAME"
 		for i in ${conflicts[@]}; do
 			echo "o$i" >> .META
