diff --git a/scripts/mybld.in b/scripts/mybld.in
index 95e5b3f..8c33a0c 100755
--- a/scripts/mybld.in
+++ b/scripts/mybld.in
@@ -149,13 +149,13 @@ rsync_binaries () {
   if [ ! -z ${PARAM[0]} ]; then
     echo "
 rsync $options \
-$URL/$(uname -m)/$VERSION/$COLLECTION/${PARAM[0]}.*  \
+${RSYNC_URL}/$(uname -m)/$VERSION/$COLLECTION/${PARAM[0]}.*  \
 $ROOTMOUNT/$(uname -m)/$VERSION/$COLLECTION/
 "
     echo -n "  OK ? (y) : "
     answer
     rsync $options \
-$URL/$(uname -m)/$VERSION/$COLLECTION/${PARAM[0]}.* \
+${RSYNC_URL}/$(uname -m)/$VERSION/$COLLECTION/${PARAM[0]}.* \
 $ROOTMOUNT/$(uname -m)/$VERSION/$COLLECTION/
     exit 0
   fi
@@ -165,7 +165,7 @@ $ROOTMOUNT/$(uname -m)/$VERSION/$COLLECTION/
 sudo mkdir -p $ROOTMOUNT/$(uname -m)/$VERSION/$COLLECTION/
 sudo chown -R $(whoami):$(whoami) $ROOTMOUNT/$(uname -m)/$VERSION/$COLLECTION/
 rsync $options \
-$URL/$(uname -m)/$VERSION/$COLLECTION/ \
+${RSYNC_URL}/$(uname -m)/$VERSION/$COLLECTION/ \
 $ROOTMOUNT/$(uname -m)/$VERSION/$COLLECTION/
 "
     echo -n "  OK ? (y) : "
@@ -173,7 +173,7 @@ $ROOTMOUNT/$(uname -m)/$VERSION/$COLLECTION/
     sudo mkdir -p $ROOTMOUNT/$(uname -m)/$VERSION/$COLLECTION/
     sudo chown -R $(whoami):$(whoami) $ROOTMOUNT/$(uname -m)/$VERSION/$COLLECTION/
     rsync $options \
-$URL/$(uname -m)/$VERSION/$COLLECTION/ \
+${RSYNC_URL}/$(uname -m)/$VERSION/$COLLECTION/ \
 $ROOTMOUNT/$(uname -m)/$VERSION/$COLLECTION/
     exit 0
   fi
@@ -182,7 +182,7 @@ $ROOTMOUNT/$(uname -m)/$VERSION/$COLLECTION/
 mkdir -p $ROOTMOUNT/$(uname -m)/$VERSION/
 sudo chown -R $(whoami):$(whoami) $ROOTMOUNT/$(uname -m)/$VERSION/
 rsync $options \
-$URL/$(uname -m)/$VERSION/ \
+${RSYNC_URL}/$(uname -m)/$VERSION/ \
 $ROOTMOUNT/$(uname -m)/$VERSION/
 "
   echo -n "  OK ? (y) : "
@@ -190,7 +190,7 @@ $ROOTMOUNT/$(uname -m)/$VERSION/
   mkdir -p $ROOTMOUNT/$(uname -m)/$VERSION/
   sudo chown -R $(whoami):$(whoami) $ROOTMOUNT/$(uname -m)/$VERSION/
   rsync $options \
-$URL/$(uname -m)/$VERSION/ \
+${RSYNC_URL}/$(uname -m)/$VERSION/ \
 $ROOTMOUNT/$(uname -m)/$VERSION/
   exit 0
 }
@@ -520,6 +520,7 @@ fi
 [ -z "$EDITOR" ] && EDITOR="vi"
 [ -z "$COLLECTIONS" ] && COLLECTIONS=(gui cli base)
 [ -z "$VERSION" ] && VERSION="$(git branch --show-current)"
+RSYNC_URL="$URL"
 
 GIT_PROJ=$(pwd)
 GIT="mybld/$(basename $GIT_PROJ)"
diff --git a/src/create.cxx b/src/create.cxx
index 3afab56..19bae40 100644
--- a/src/create.cxx
+++ b/src/create.cxx
@@ -24,6 +24,8 @@ create::create(CardsArgumentParser& argParser)
     m_tree = m_pkgfile.getListOfPackages();
     buildDatabase(true);
 
+	readRulesFile();
+
     if (!m_argParser.isSet(CardsArgumentParser::OPT_DRY)) {
         core();
         chdir(m_portsDir.c_str());
@@ -248,8 +250,6 @@ void create::run()
 	// Reading the archiving to find a list of files
 	std::pair<std::string, cards::db> package = openArchivePackage(m_packageArchiveName);
 
-	readRulesFile();
-
 	std::set<std::string> non_install_files = applyInstallRules(package.first,
 		package.second, m_actionRules);
 
diff --git a/src/pkgdbh.cxx b/src/pkgdbh.cxx
index 4c14ba1..34970d1 100644
--- a/src/pkgdbh.cxx
+++ b/src/pkgdbh.cxx
@@ -1309,9 +1309,8 @@ void pkgdbh::installArchivePackage
 	}
 	FREE_ARCHIVE(archive);
 }
-void pkgdbh::readRulesFile()
+const void pkgdbh::readRulesFile()
 {
-	m_actionRules.clear();
 	unsigned int linecount = 0;
 	const std::string filename = m_root + PKGADD_CONF;
 	std::ifstream in(filename.c_str());
diff --git a/src/pkgdbh.h b/src/pkgdbh.h
index 0485672..d671b39 100644
--- a/src/pkgdbh.h
+++ b/src/pkgdbh.h
@@ -83,7 +83,7 @@ protected:
      */
     void                    moveMetaFilesPackage(const std::string& name, cards::db& info);
 
-    void                    readRulesFile();
+    const void              readRulesFile();
     void                    getInstallRulesList(const std::vector<rule_t>& rules,
                                 cards::rule_event_t event, std::vector<rule_t>& found) const;
 
