diff --git a/scripts/mybld.in b/scripts/mybld.in
index 9d1b678..9fd8371 100755
--- a/scripts/mybld.in
+++ b/scripts/mybld.in
@@ -159,8 +159,8 @@ $ROOTMOUNT/$(uname -m)/$VERSION/$COLLECTION/
 "
     echo -n "  OK ? (y) : "
     answer
-    sudo chown -R $(whoami):$(whoami) $ROOTMOUNT/$(uname -m)$VERSION/$COLLECTION/${PARAM[0]}.*
-    rm $ROOTMOUNT/$(uname -m)$VERSION/$COLLECTION/${PARAM[0]}.*
+    sudo chown -R $(whoami):$(whoami) $ROOTMOUNT/$(uname -m)/$VERSION/$COLLECTION/${PARAM[0]}.*
+    rm $ROOTMOUNT/$(uname -m)/$VERSION/$COLLECTION/${PARAM[0]}.*
     rsync $options \
 $URL/$(uname -m)/$VERSION/$COLLECTION/${PARAM[0]}.* \
 $ROOTMOUNT/$(uname -m)/$VERSION/$COLLECTION/
