diff --git a/src/create.cxx b/src/create.cxx
index ccdf94a..8776003 100644
--- a/src/create.cxx
+++ b/src/create.cxx
@@ -23,6 +23,8 @@ create::create(CardsArgumentParser& argParser)
 
     core();
 
+    chdir(m_portsDir.c_str());
+
     if (isACollection())
         buildCollection();
     else
diff --git a/src/pkgrepo.cxx b/src/pkgrepo.cxx
index 918ffcf..e18230a 100644
--- a/src/pkgrepo.cxx
+++ b/src/pkgrepo.cxx
@@ -167,24 +167,27 @@ void pkgrepo::generateDependencies()
             }
 
             // archive PackageName exist:
-            // Let's see if we need to download it.
-            // If the binary archive is not yet downloaded
-            if (!checkFileExist(packageFileName))
-                downloadPackageFileName(packageName);
-
-            // m_packageFileNameHash contains hashsum of packageFileName
-            generateHash(packageFileName);
-
-            // we compare m_packageFileNameHash with the one stored in the database
-            if (!checkHash(packageName)) {
-                // Last try to get it.
-                downloadPackageFileName(packageName);
-            }
-            if (!checkSign(packageName)) {
-                m_actualError = cards::ERROR_ENUM_WRONG_SIGNATURE;
-                treatErrors(packageName);
+            // Let's see if we need to download it
+            // if the binary archive is not yet downloaded
+            // Make no sense to download if no url exist
+            if (m_config.url().size() > 0) {
+                if (!checkFileExist(packageFileName))
+                    downloadPackageFileName(packageName);
+                // m_packageFileNameHash contains
+                // the hashsum of packageFileName
+                generateHash(packageFileName);
+
+                // we compare m_packageFileNameHash
+                // with the one stored in the database
+                if (!checkHash(packageName)) {
+                    // Last try to get it.
+                    downloadPackageFileName(packageName);
+                }
+                if (!checkSign(packageName)) {
+                    m_actualError = cards::ERROR_ENUM_WRONG_SIGNATURE;
+                    treatErrors(packageName);
+                }
             }
-
             directDependencies = getPackageDependencies(packageFileName);
         }
 
