diff --git a/scripts/pkgmk.in b/scripts/pkgmk.in
index 96dee2a..e941a18 100755
--- a/scripts/pkgmk.in
+++ b/scripts/pkgmk.in
@@ -76,7 +76,7 @@ get_collections() {
 }
 get_filename() {
 	if [[ $1 =~ git$ ]]; then
-		echo "$PKGMK_SOURCE_DIR/$name-$version${PKGMK_GIT_EXT}"
+		echo "$PKGMK_SOURCE_DIR/$name${PKGMK_GIT_EXT}"
 	elif [[ $1 =~ ^(http|https|ftp|file)://.*/(.+) ]]; then
 		echo "$PKGMK_SOURCE_DIR/${BASH_REMATCH[2]}"
 	else
@@ -203,12 +203,9 @@ download_file() {
 				error "please install or add git in the dependencies."
 				exit ${E_DOWNLOAD}
 			fi
-			git clone $1 $name-$version
-			cd $name-$version
-			git checkout $version
-			cd -
-			bsdtar -a -cvf $PKGMK_SOURCE_DIR/$name-$version${PKGMK_GIT_EXT} $name-$version
-			rm -rf $name-$version
+			git clone $1 $name
+			bsdtar -a -cvf $PKGMK_SOURCE_DIR/$name${PKGMK_GIT_EXT} $name
+			rm -rf $name
 			return ;;
 		*)
 
@@ -293,6 +290,22 @@ download_source() {
 					exit ${E_DOWNLOAD}
 				fi
 			fi
+		else
+			case $LOCAL_FILENAME in
+				*.git)
+					if [ ! "`type -p git`" ]; then
+						error "please install or add git in the dependencies."
+						exit ${E_DOWNLOAD}
+					fi
+					bsdtar -p -o -C $SRC -xf $LOCAL_FILENAME
+					cd $name
+					git pull
+					cd -
+					bsdtar -a -cvf $PKGMK_SOURCE_DIR/$name${PKGMK_GIT_EXT} $name
+					rm -rf $name
+					return;;
+				*)
+			esac
 		fi
 	done
 }
