diff --git a/src/create.cxx b/src/create.cxx
index 1ced6a7..354155f 100644
--- a/src/create.cxx
+++ b/src/create.cxx
@@ -171,6 +171,7 @@ void create::installDependencies(std::string& packageName)
         ++currentLevel;
     }
     if (! m_dependencies.empty()) {
+        readRulesFile();
         for (auto i : m_dependencies) {
             if ( i == packageName)
                 break;
@@ -224,6 +225,18 @@ void create::buildCollection()
             buildBinary(i);
             continue;
         }
+        if (!checkFileExist(pkgrepo.dirName(i)
+            + "/"
+            + pkgrepo.fileName(i))) {
+            std::cout << i
+                << " ===> SHOULD BE BUILD !"
+                << std::endl;
+
+            if (m_argParser.isSet(CardsArgumentParser::OPT_DRY))
+                continue;
+            buildBinary(i);
+            continue;
+        }
         if (pkgrepo.version(i) != pkgfile.getPortVersion(i)) {
             std::cout << i
                 <<": Binary: "
diff --git a/src/pkgfile.cxx b/src/pkgfile.cxx
index 47ee998..c2fba3d 100644
--- a/src/pkgfile.cxx
+++ b/src/pkgfile.cxx
@@ -262,20 +262,31 @@ std::vector<std::string>& pkgfile::pkgfile::getListOfPackagesFromCollection(std:
     if (m_listOfPackages.size() == 0)
         parse();
 
+    unsigned int level, levelMax;
+    level = 0;
+    levelMax = 0;
+
     bool found = false;
-    unsigned int level = 0;
     while (true) {
+        for (auto i : m_listOfPackages)
+            if (i.second.level() == level)
+                found = true;
+        if (!found)
+            break;
+
+        ++level;
+        found = false;
+    }
+    levelMax = level;
+    level = 0;
+    while (level <= levelMax) {
         for (auto i : m_listOfPackages) {
             if ((i.second.collection() == collectionName) &&
                 (i.second.level() == level)) {
                 m_listOfPackagesFromCollection.push_back(i.first);
-                found = true;
             }
         }
-        if (!found)
-            break;
         level++;
-        found = false;
     }
     return m_listOfPackagesFromCollection;
 }
