diff --git a/scripts/pkgmk.in b/scripts/pkgmk.in
index 68dba7b..6e2dc1a 100755
--- a/scripts/pkgmk.in
+++ b/scripts/pkgmk.in
@@ -741,6 +741,7 @@ add_meta_to_archive() {
 	echo "r$release" >> .META
 	echo "S$SIZE_I" >> .META
 	echo "c$COLLECTION" >> .META
+	echo "a$PKGMK_ARCH" >> .META
 
 	if [ -z "$GROUP" ]; then
 		echo "D$DESC" >> .META
@@ -749,7 +750,6 @@ add_meta_to_archive() {
 		echo "M$MAINT" >> .META
 		echo "P$PACK" >> .META
 		echo "L$LICENSE" >> .META
-		echo "a$PKGMK_ARCH" >> .META
 		echo "B$PKGMK_BUILDVER" >> .META
 	else
 	    echo "g$GROUP" >> .META
diff --git a/src/create.cxx b/src/create.cxx
index 6f4d4e1..c48c5d8 100644
--- a/src/create.cxx
+++ b/src/create.cxx
@@ -28,7 +28,7 @@ create::~create()
 void create::base()
 {
     std::vector<std::string> list;
-    buildDatabase(false, false);
+    buildDatabase(false, true);
     for ( auto i : m_listOfPackages) {
         if (i.second.collection() == "base")
             continue;
@@ -39,17 +39,11 @@ void create::base()
         m_packageName = i;
         cards::lock Lock(m_root, true);
 
-	    // Get the list of installed packages
-	    getListOfPackagesNames(m_root);
-
-	    // Retrieve info about all the packages
-	    buildDatabase(false, true);
-
         // Remove metadata about the package removed
-	    removePackageFilesRefsFromDB(m_packageName);
+        removePackageFilesRefsFromDB(m_packageName);
 
         // Remove the files on hd
-	    removePackageFiles(m_packageName);
+        removePackageFiles(m_packageName);
 
     }
 }
@@ -160,7 +154,7 @@ void create::installDependencies(std::string& packageName)
         ++currentLevel;
     }
     if (!m_dependencies.empty()) {
-        buildDatabase(false, false);
+
         for (auto i : m_dependencies) {
             std::string name;
             if ( i == packageName)
@@ -425,7 +419,7 @@ void create::buildBinary(std::string packageName)
             + " "
             + version;
         m_upgrade = 0;
-        buildDatabase(false, false);
+
 		if (checkPackageNameExist(name)) {
 			message = name + ": is ALLREADY installed";
 			m_upgrade = 1;
diff --git a/src/install.cxx b/src/install.cxx
index 916ca9d..036b0b1 100644
--- a/src/install.cxx
+++ b/src/install.cxx
@@ -23,7 +23,7 @@ install::install(const CardsArgumentParser& argParser,
 		}
 	}
 
-	buildDatabase(false, false);
+	buildDatabase(false, true);
 	for(auto pkg : m_argParser.otherArguments()) {
 		std::set<std::string> SetOfPackages = m_pkgrepo.getListOfPackagesFromSet(pkg);
 		if (SetOfPackages.empty())
diff --git a/src/pkgdbh.cxx b/src/pkgdbh.cxx
index de75f04..b0aa750 100644
--- a/src/pkgdbh.cxx
+++ b/src/pkgdbh.cxx
@@ -10,14 +10,12 @@ using __gnu_cxx::stdio_filebuf;
 pkgdbh::pkgdbh(const std::string& name)
 	: m_utilName(name)
 	, m_DB_Empty(true)
-	, m_miniDB_Empty(true)
 {
 	openlog(m_utilName.c_str(),LOG_CONS,LOG_LOCAL7);
 }
 pkgdbh::pkgdbh()
 	: m_utilName("pkgdbh")
 	, m_DB_Empty(true)
-	, m_miniDB_Empty(true)
 	, m_root("/")
 {
 }
diff --git a/src/pkgdbh.h b/src/pkgdbh.h
index 6e7f8fe..2464ec2 100644
--- a/src/pkgdbh.h
+++ b/src/pkgdbh.h
@@ -36,7 +36,6 @@ class pkgdbh {
 
     bool                    m_dependency;
     bool                    m_DB_Empty;
-    bool                    m_miniDB_Empty;
 
     void                    runLastPostInstall();
     std::string             getSingleItem(const std::string& PackageName, const char i) const;
